Initial commit

jmeloranta opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

jmeloranta commented:
My first flathub submission - so things are most likely messed up.... :-)

barthalion commented:
bot, build net.crossfireatlas.crossfire

barthalion commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,26 @@
+        sha256: e6af744ea986da0991233787be6c9bc5535a7f3fe8ea682fe3284f1152e67b77
+      - type: file
+        path: net.crossfireatlas.crossfire.start.sh
+finish-args: 
```
`finish-args` are typically above `modules`

hfiguiere replied:
also missing `--share=ipc` which is required for x11. (I thought this was an error in the linter @barthalion ?)

jmeloranta replied:
OK, thanks. I am adding this.

hfiguiere replied:
this is not completely addressed. see https://github.com/flathub/flathub/pull/3757/files#r1058270721

hfiguiere replied:
I'll reiterate what was said:

> `finish-args` are typically above `modules`

barthalion commented on net.crossfireatlas.crossfire.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
You need to explicitly reference this file in your manifest and install it at correct place.

jmeloranta replied:
OK, I have added installing .desktop and .appdata.

barthalion commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
You need to explicitly reference this file in your manifest and install it at correct place.

barthalion commented on README.md:
```diff
@@ -0,0 +1,34 @@
+# Flathub: Crossfire rpg
```
There's generally no need to have any README here, as it will not be seen by any user.

jmeloranta replied:
OK but since I put one together, it might as well be there.

barthalion commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,35 @@
+	<project_license>GPL-2.0</project_license>
+	<!-- Description taken from https://crossfire.real-time.com -->
+	<description>
+                <p>An introduction to the game of Crossfire, and links to other sites and what they have to say about the game.</p>
+                <p>In the words of its original author Frank Tore Johansen:</p>
```
Does it make sense to make this a quote?

jmeloranta replied:
The quote kind of describes the game (as the original author put it). But I guess it can be stated without quoting anyone. So, removed the quote part.

barthalion commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,35 @@
+                <p>"Crossfire is a multi-player graphical arcade and adventure game made for the X Windows systems environment. It has certain flavours from other games, especially gauntlet and the rogue-like games (Nethack, Moria, Angband, and Ragnarok.) Any number of players can move around in their own window, finding and using items and battling monsters. They can choose to cooperate or compe ...
+                <p>In the years of development that have followed, Crossfire has grown to encompass over 150 monsters, about 3000 maps to explore, an elaborate magic system, over 15 character types, a system of skills, and many, many artifacts and treasures.</p>
+	</description>
+	<developer_name>Multiple authors, see http://crossfire.real-time.com/</developer_name>
```
I'd change it to "Crossfire developers".

jmeloranta replied:
Done.

barthalion commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,26 @@
+      - install -D net.crossfireatlas.crossfire.start.sh /app/bin/net.crossfireatlas.crossfire.start.sh
+    sources:
+      - type: file
+        url: https://sourceforge.net/projects/crossfire/files/jxclient/jxclient.jar/download
```
This URL looks unversioned. Is it on purpose?

jmeloranta replied:
That is how it comes from sourceforge. So, yes.

barthalion commented on net.crossfireatlas.crossfire.start.sh:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+/app/jre/bin/java -jar /app/bin/jxclient.jar
+exit 0
```
```suggestion
exec /app/jre/bin/java -jar /app/bin/jxclient.jar
```

jmeloranta replied:
Done.

barthalion commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,26 @@
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions: 
```
I assume it isn't meant to be empty.

hfiguiere replied:
```
(flatpak-builder:2): Json-WARNING **: 11:11:30.531: Failed to deserialize "sdk-extensions" property of type "GStrv" for an object of type "BuilderManifest"
```

jmeloranta replied:
I have added: org.freedesktop.Sdk.Extension.openjdk11

hfiguiere commented on net.crossfireatlas.crossfire.desktop:
```diff
@@ -0,0 +1,9 @@
+Categories=Game;
+Comment=Roguelike multi-user RPG
+Exec=net.crossfireatlas.crossfire.start.sh
+Icon=net.crossfireatlas.crossfire.ico
```
1. .ico isn't a valid icon format. This is not MS-DOS
2. It's the *NAME*. No extension.

jmeloranta replied:
OK, .ico removed.

hfiguiere commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,26 @@
+  - name: openjdk
+    buildsystem: simple
+    build-commands: 
+      - /usr/lib/sdk/openjdk11/install.sh
```
it's actually surprising the build didn't fail beucase this is missing. See comment about about the empty sdk-extension.

jmeloranta replied:
I added sdk-extensions: org.freedesktop.Sdk.Extension.openjdk11
to the yaml file. So, this resolves this comment?

jmeloranta commented:
I have no pushed all commits forward. Hopefully we are converging.. Sorry about the issues - this is first flathub submission for me.

hfiguiere commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,30 @@
+      - install -D download /app/bin/jxclient.jar
+      - install -D net.crossfireatlas.crossfire.start.sh /app/bin/net.crossfireatlas.crossfire.start.sh
+      - install -D net.crossfireatlas.crossfire.desktop /app/share/applications/net.crossfireatlas.crossfire.desktop
+      - install -D net.crossfireatlas.crossfire.appdata.xml /app/share/metainfo/net.crossfireatlas.crossfire.appdata.xml
```
you still don't install an icon.

And as I said, `.ico` isn't a valid icon format. PNG (128x128 at least)  or SVG are only accepted.

jmeloranta replied:
OK, I have now the PNG (128x128) included (and references in the yaml file).

hfiguiere commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,33 @@
+	<id>net.crossfireatlas.crossfire</id>
+	<name>Crossfire RPG</name>
+	<summary>Roguelike multi-user RPG</summary>
+	<metadata_license>CC-BY-SA-3.0</metadata_license>
```
CC0-1.0 is prefered for the metadata.

jmeloranta replied:
OK, done.

hfiguiere commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,33 @@
+	<url type="homepage">https://crossfire.real-time.com/</url>
+	<screenshots>
+		<screenshot type="default">
+			<image>https://raw.githubusercontent.com/flathub/net.crossfireatlas.crossfire/master/screenshot1.png</image>
```
this is not a screenshot, this is a banner / logo.

jmeloranta replied:
OK, yes, that is correct. I removed the banner and left the screenshot (one).

hfiguiere commented:
I know it will fail:

bot, build net.crossfireatlas.crossfire

hfiguiere commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,30 @@
+        url: https://sourceforge.net/projects/crossfire/files/jxclient/jxclient.jar/download
+        sha256: e6af744ea986da0991233787be6c9bc5535a7f3fe8ea682fe3284f1152e67b77
+      - type: file
+        path: net.crossfireatlas.crossfire.start.sh
```
the files installed above still are not referenced here. .desktop, .appdata.xml and icon in the right format.

This is the build failure atm.

jmeloranta replied:
OK, done.

hfiguiere commented:
bot, build net.crossfireatlas.crossfire

hfiguiere commented on net.crossfireatlas.crossfire.yaml:
```diff
@@ -0,0 +1,37 @@
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions: org.freedesktop.Sdk.Extension.openjdk11
```
```suggestion
sdk-extensions: 
  - org.freedesktop.Sdk.Extension.openjdk11
```

hfiguiere replied:
```
(flatpak-builder:2): Json-WARNING **: 20:48:15.715: Failed to deserialize "sdk-extensions" property of type "GStrv" for an object of type "BuilderManifest"
```

hfiguiere commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,30 @@
+			<image>https://raw.githubusercontent.com/flathub/net.crossfireatlas.crossfire/master/screenshot.png</image>
+		</screenshot>
+	</screenshots>
+	<releases>
+		<release version="1.0" date="2022-11-19">
+			<description>
+				<p>v1.0 Crossfire java client.</p>
+			</description>
+	</releases>
```
```suggestion
	<releases>
		<release version="1.0" date="2022-11-19">
			<description>
				<p>v1.0 Crossfire java client.</p>
			</description>
		</release>
	</releases>
```

hfiguiere commented on net.crossfireatlas.crossfire.appdata.xml:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>net.crossfireatlas.crossfire</id>
```
there doesn't seem to be a working `crossfireatlas.net` and no mention of it here.

App-id as supposed to reference a domain controlled by the developers.

jmeloranta replied:
There is crossfireatlas.net and it works (just tested):
http://www.crossfireatlas.net/
So, I am not sure what the problem is?

jmeloranta replied:
Not sure what happened to my response. Anyway:
crossfireatlas.net exists. You can check:
http://www.crossfireatlas.net/
So, not sure what the problem is?

hfiguiere replied:
no problem. it didn't seem to work when I checked.

jmeloranta replied:
We decided (per discussion on the crossfire developer mailing list) to use crossfireatlas.net rather than crossfire.real-time.com because of the dash (flatpak does not seem to like dashes).

jmeloranta commented:
I wonder if there are any more open problems?

hfiguiere commented:
You have ignored this THREE TIMES (only the first occurence shown)

![image](https://user-images.githubusercontent.com/114441/210023382-703e0afd-dc79-4a32-9442-91c813c7eb59.png)

jmeloranta commented:
> You have ignored this THREE TIMES (only the first occurence shown)

Well, you said *typically* implying that is not a requirement. I have moved finish-args before the modules now. U happy now?

barthalion commented:
bot, build net.crossfireatlas.crossfire

jmeloranta commented:
OK, as the buildbot suggests, I have removed the URL from developer in the app file. I changed the screenshot URL to point to my flathub page, I am not sure this is the right way to do it? It is now: 
https://github.com/jmeloranta/flathub/raw/net.crossfireatlas.crossfire/screenshot.png
Any suggestions?

barthalion commented:
Yes, that's good for now. You should switch it to `flathub/flathub` after it's merged.

bot, build net.crossfireatlas.crossfire

barthalion commented:
LGTM. To answer some questions from the mailing list discussion: while runtimes don't ship gtk2 any more, it's available in https://github.com/flathub/shared-modules/ (see the `gtk2` subdirectory). The domain with dashes can be used, but dashes need to be replaced with underscore, except for the last segment. Should I merge it as-is, or would you like to switch domain or to the GTK2 client?

jmeloranta commented:
Let's merge it as it is now. I may work on including the gtk2 client in the future (but that code should really be rewritten for gtk3/4). The domain names were OK with all the crossfire developers, so let's keep that as it is to avoid issues with dashes. Thank you for your help.

barthalion commented:
/merge
